### PR TITLE
Remove bitcoincore rpc client

### LINKED ISSUE #48: bitcoincore-rpc client is not async
## Problem: 
The bitcoincore-rpc client we are using is not async and hence blocks the current thread during execution. This is not good because the remaining code is based on futures and hence the executor cannot do its work.

## Goal: 
Don't block the executor thread.

## Recommendation:
~We only use the client to generate blocks and fund.~

~Remove the dependency on the client and handcraft the JSON-rpc request with an async HTTP client like reqwest.~

Look into building an image on the fly using shiplift crate that:
- Generate 101 blocks
- Do the funding
- Update contrab to do regular block generation

### PR DESCRIPTION
**Pull request by D4nte**
Resolves #48

### CODE REVIEW

File: src/docker/blockchain/bitcoin.rs [hunk 1]
```diff
@@ -58,36 +59,95 @@ impl Image for BitcoinNode {
...
+            id: "fund".to_string(),
+            method: "sendtoaddress".to_string(),
+            params: vec![address.to_string(), amount.as_btc().to_string()],
+        }
     }
 }
 
+#[derive(Debug, serde::Deserialize)]
+struct FundResponse {
+    result: sha256d::Hash,
```
Comment (luckysori) [inline src/docker/blockchain/bitcoin.rs]: :wrench: 
```suggestion
    txid: sha256d::Hash,
```

File: src/docker/blockchain/bitcoin.rs [hunk 2]
```diff
@@ -58,36 +59,95 @@ impl Image for BitcoinNode {
...
-
-        let response = client.generate(101, None).and_then(|_| {
-            client
-                .send_to_address(&address, value, None, None, None, None, None, None)
-                .and_then(|txid| client.generate(1, None).map(|_| txid))
-        });
+        amount: Self::Amount,
+    ) -> Box<dyn Future<Item = Self::TxId, Error = Self::ClientError> + Send> {
+        let generate_req = GenerateQuery::new(101);
+
```
Comment (luckysori) [inline src/docker/blockchain/bitcoin.rs]: :upside_down_face: Unintended newline?

File: src/docker/blockchain/bitcoin.rs [hunk 3]
```diff
@@ -58,36 +59,95 @@ impl Image for BitcoinNode {
...
+                let password = Some(self.password.clone());
+
+                move |_| {
+                    Client::new()
+                        .post(&endpoint)
+                        .basic_auth(username, password)
+                        .json(&fund_req)
+                        .send()
+                        .and_then(|mut response| response.json::<FundResponse>())
+                        .and_then(|response| Ok(response.result))
```
Comment (luckysori) [inline src/docker/blockchain/bitcoin.rs]: :wrench: 
```suggestion
                        .and_then(|response| Ok(response.txid))
```

File: src/docker/blockchain/bitcoin.rs (hunk 1 above)
Comment (D4nte) [reply src/docker/blockchain/bitcoin.rs]: This is what bitcoind returns! I have no choice in the naming.

**Pull request merged**